core/lib/common/worker.js

```diff
@@ -261,11 +312,139 @@ module.exports = class Worker {
+	async createTunneltoDUT(
+		target,
+		dutPort,
+		workerPort
+	) {
+		let ip =  await this.getDutIp(target);
+		//get containerID of the worker container on the testbot - we must do socat from inside the worker container
+		let containerId = await this.executeCommandInWorkerHost(`balena ps | grep worker | awk '{print $1}'`);
+		let argsWorker = [];
+		argsWorker = argsWorker.concat([
+			`${this.workerUser}@${this.workerHost}`,
+			`-p`,
+			this.workerPort,
+			`-i`,
+			this.sshKey,
+			`-o`,
+			`StrictHostKeyChecking=no`,
+			`-o`,
+			`UserKnownHostsFile=/dev/null`,
+		]);
+
+		if(this.sshPrefix != ''){
+			argsWorker = argsWorker.concat([
+				`host`,
+				this.uuid
+			])
+		}
+
+		argsWorker = argsWorker.concat([
+			`balena`,
+			`exec`,
+			containerId,
+			`socat`,
+			`tcp-listen:${workerPort},reuseaddr,fork "system:ssh ${ip} -p 22222 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${this.dutSshKey} /usr/bin/nc localhost ${dutPort}"`
+		]);
+
+		let tunnelProWorker = spawn(`ssh`, argsWorker);
+
+		// setup a listener from this host to worker must be a sub process... 
+		// we must give map the same port on this host and the DUT - so the cli can use it 
+		// this will be torn down at the end of the tests when the core is destroyed
+		let argsClient = [
+			`tcp-listen:${dutPort},reuseaddr,fork`,
+			`system:ssh ${this.workerUser}@${this.workerHost} -p ${this.workerPort} -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${this.sshKey} ${this.sshPrefix}/usr/bin/nc localhost ${workerPort}`
+		];
+
+		let tunnelProcClient = spawn(`socat`, argsClient);
+	}
+
+	// create tunnels to relevant DUT ports to we can access them remotely
+	async createSSHTunnels(target){
+		if(!this.url.includes(`localhost`)){
+			const DUT_PORTS = [
+				48484, // supervisor
+				22222, // ssh
+				2375 // engine
+			]
+			let workerPort = 8888;
+			for(let port of DUT_PORTS){
+				console.log(`creating tunnel to dut port ${port}...`)
+				await this.createTunneltoDUT(
+					target,
+					port,
+					workerPort
+				)
+				workerPort = workerPort + 1;
+			}
+		} 
+	}
+
+    // sends file over rsync to worker container
+	async rsyncSendFiletoWorker(filePath, destination){
+		let containerId = await this.executeCommandInWorkerHost(`balena ps | grep worker | awk '{print $1}'`);
+		// todo : replace with npm package
+		await exec(`rsync -av -e "ssh ${this.workerUser}@${this.workerHost} -p ${this.workerPort} -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -q ${this.sshPrefix}balena exec -i" ${filePath} ${containerId}:${destination}`);
+	}
+
+	// sends file over rsync to DUT host os
+	async rsyncSendFiletoDUT(filePath, destination, target){
+		// todo : replace with npm package
+		let ip = await this.ip(target);
+		await exec(`rsync -av -e "ssh -p 22222 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -q -i ${this.sshKey}" ${filePath} root@${ip}:${destination}`);
+	}
```

Comment:
Are these functions meant to be internal, or used from test suites as well? If they're internal only, I think they're fine. If they're meant to be used from suites, I see a number of problems.

Reply:
At the moment, I'm only using the sendFileToWorker internally.

The send file to DUT I'm using in the HUP suite:  https://github.com/balena-os/meta-balena/blob/ryan/tests-update-for-leviathan/tests/suites/hup/suite.js#L91

Reply:
@rcooke-warwick If these functions are for internal use only, they can be easily changed, and my feedback is moot. If they're used by the test suites, they're an interface, and changing them will be breaking, so we'll want to get it right the first time, if possible.

Rsync is an implementation detail, I don't see any value to including it in the function name. If anything, it'll confuse things in the future if we change the implementation.

In fact, worker/DUT also appears at first glance to be a parameter, I don't currently understand the need for separate functions here. At the end of the day, we're transferring files from one host to another. Which host does not need to be encoded in the function name.